File: slack-reader/src/test/java/dev/number6/slackreader/dagger/FakeConfigurationPortModule.java
package dev.number6.slackreader.dagger;

import dev.number6.slackreader.adaptor.EnvironmentVariableSlackReaderConfigurationAdapter;
import dev.number6.slackreader.port.SlackReaderConfigurationPort;
import dagger.Module;
import dagger.Provides;

@Module
public class FakeConfigurationPortModule {

    @Provides
    public SlackReaderConfigurationPort slackReaderConfigurationPort() {
        return new EnvironmentVariableSlackReaderConfigurationAdapter();
    }
}

File: slack-reader/src/test/kotlin/dev/number6/slackreader/dagger/FakeConfigurationPortModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slackreader.port.SlackReaderConfigurationPort

@Module
class FakeConfigurationPortModule {
    @Provides
    fun slackReaderConfigurationPort(): SlackReaderConfigurationPort {
        return object: SlackReaderConfigurationPort {
            override fun getTopicArn(): String {
                return "TOPIC_ARN"
            }

            override fun getBlacklistedChannels(): Collection<String> {
                return listOf("blacklisted", "channels")
            }

        }
    }
}